Reformatted `XorBlendOp` (as a pilot for subsequent changes)

Lehonti opened the pull request:
I moved it from `UserBlendOps.Generated.cs` to its own file, and formatted it so that the long methods are not made of single lines.

I want this to serve as a model for the rest of the blend operations, and then (when the code is very neat) we could probably copy them to [the Pinta image manipulation library](https://github.com/PintaProject/Pinta.ImageManipulation) (depending on whether that project has a future).

One thing I noticed, though is that the constructor of `XorBlendOpWithOpacity` does a range check on the _field_ instead of the _argument_. Since this is most likely a mistake, I fixed it too (see second commit).

cameronwhite commented:
This code is originally from paint.net, so it'll need to preserve the license header from the original file
I'm fine with this reformatting being done in bulk, but I'd caution against spending further time refactoring this code since it's hardly ever touched - I'd rather see improvements to active areas of the codebase, especially areas which need bug fixes or enhancements

Lehonti commented:
In that case I won't spend more time on this unless it's needed. I just moved the reformatted version to its original file, and deleted the new file. In case we ever finish the reformatting we'd have this pull request as reference
